### PR TITLE
Issue #441: Fix dataChange flag in optimize

### PR DESCRIPTION
**Pull request by JosepSampe**
## Description

- Fixes the dataChange flag issue #441 during `optimize()`

## Type of change

- Bug fix

## Checklist:

Here is the list of things you should do before submitting this pull request:

- [x] New feature / bug fix has been committed following the [Contribution guide](https://github.com/Qbeast-io/qbeast-spark/blob/main/CONTRIBUTING.md).
- [ ] Add logging to the code following the [Contribution guide](https://github.com/Qbeast-io/qbeast-spark/blob/main/CONTRIBUTING.md).
- [ ] Add comments to the code (make it easier for the community!).
- [ ] Change the documentation.
- [x] Add tests.
- [x] Your branch is updated to the main branch (dependent changes have been merged).

## How Has This Been Tested? (Optional)

Included a unit test

### CODE REVIEW

**Review by osopardo1 — APPROVED** [review]
lgtm

File: src/test/scala/io/qbeast/spark/utils/QbeastOptimizeIntegrationTest.scala [hunk 1]
```diff
@@ -107,4 +111,33 @@ class QbeastOptimizeIntegrationTest extends QbeastIntegrationTestSpec {
...
+
+      val deltaLog = DeltaLog.forTable(spark, tmpDir)
+      val snapshot = deltaLog.update()
+      val conf = deltaLog.newDeltaHadoopConf()
+
+      val fileActions = deltaLog.store
+        .read(FileNames.deltaFile(deltaLog.logPath, snapshot.version), conf)
+        .map(Action.fromJson)
+
+      fileActions.foreach {
```
Comment (Jiaweihu08) [inline src/test/scala/io/qbeast/spark/utils/QbeastOptimizeIntegrationTest.scala:135]: You can also check that the `CommitInfo` has `SnapshotIsolation` as `isolationLevel`?

File: src/main/scala/io/qbeast/spark/table/IndexedTable.scala [hunk 2]
```diff
@@ -540,7 +541,10 @@ private[table] class IndexedTableImpl(
           val (dataExtended, tableChanges) =
             DoublePassOTreeDataAnalyzer.analyzeOptimize(data, indexStatus)
 
-          val newFiles = dataWriter.write(tableID, schema, dataExtended, tableChanges)
+          val newFiles = dataWriter
+            .write(tableID, schema, dataExtended, tableChanges)
+            .map(addFile => addFile.asInstanceOf[AddFile].copy(dataChange = false))
```
Comment (Jiaweihu08) [inline src/main/scala/io/qbeast/spark/table/IndexedTable.scala:546]: Pattern matching is the more appropriate and idiomatic approach in scala than casting with `asInsatanceOf:`
```
.collect {
  case addFile: AddFile => addFile.copy(dataChange = false)
}
```

Comment (Jiaweihu08) [inline src/main/scala/io/qbeast/spark/table/IndexedTable.scala:546]: Pattern matching is the more appropriate and idiomatic approach in scala than casting with `asInsatanceOf:`
```
.collect {
  case addFile: AddFile => addFile.copy(dataChange = false)
}
```

File: src/test/scala/io/qbeast/spark/utils/QbeastOptimizeIntegrationTest.scala (hunk 1 above)
Comment (Jiaweihu08) [inline src/test/scala/io/qbeast/spark/utils/QbeastOptimizeIntegrationTest.scala:135]: Can you also check that the `CommitInfo` has `SnapshotIsolation` as `isolationLevel`?

**Review by Jiaweihu08 — CHANGES_REQUESTED** [review]

File: src/test/scala/io/qbeast/spark/utils/QbeastOptimizeIntegrationTest.scala (hunk 1 above)
Comment (JosepSampe) [reply src/test/scala/io/qbeast/spark/utils/QbeastOptimizeIntegrationTest.scala]: Updated

**Review by Jiaweihu08 — APPROVED** [review]

**Pull request merged**